File: generators/dndShardmindNames.py
__all__ = ['dndShardmindNames']

# Don't look below, you will not understand this Python code :) I don't.

from js2py.pyjs import *
# setting scope
var = Scope( JS_BUILTINS )
set_global_object(var)

# Code follows:
var.registers(['nm1', 'nm2', 'nameMas', 'nameGen'])
@Js
def PyJsHoisted_nameMas_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers([])
    var.put('rnd', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm1').get('length'))))
    var.put('rnd2', var.get('Math').callprop('floor', (var.get('Math').callprop('random')*var.get('nm2').get('length'))))
    var.put('nMs', (var.get('nm1').get(var.get('rnd'))+var.get('nm2').get(var.get('rnd2'))))
    var.get('testSwear')(var.get('nMs'))
PyJsHoisted_nameMas_.func_name = 'nameMas'
var.put('nameMas', PyJsHoisted_nameMas_)
@Js
def PyJsHoisted_nameGen_(this, arguments, var=var):
    var = Scope({'arguments':arguments, 'this':this}, var)
    var.registers(['result'])
    var.put('result', Js([]))
    #for JS loop
    var.put('i', Js(0.0))
    while (var.get('i')<Js(10.0)):
        try:
            var.get('nameMas')()
            while PyJsStrictEq(var.get('nMs'),Js('')):
                var.get('nameMas')()
        finally:
                (var.put('i',Js(var.get('i').to_number())+Js(1))-Js(1))
    pass
    return var.get('result')
PyJsHoisted_nameGen_.func_name = 'nameGen'
var.put('nameGen', PyJsHoisted_nameGen_)
var.put('nm1', Js([Js('Adu'), Js('Ama'), Js('Ani'), Js('Ar'), Js('Arsha'), Js('Ashi'), Js('Ashtu'), Js('Bala'), Js('Bara'), Js('Basha'), Js('Beles'), Js('Delu'), Js('Di'), Js('Dura'), Js('Duru'), Js('Enu'), Js('Eri'), Js('Eshu'), Js('Hua'), Js('Hun'), Js('Il'), Js('Ilu'), Js('Ira'), Js('Ish'), Js('Ku'), Js('Kua'), Js('Kuba'), Js('Lu'), Js('Mani'), Js('Mara'), Js('Mashi'), Js('Na'), Js('Nara'), Js('Nashi'), Js('Nu'), Js('Rua'), Js('Run'), Js('Sana'), Js('Sari'), Js('Selu'), Js('Shir'), Js('Suma'), Js('Tab'), Js('Tin'), Js('Tiru'), Js('Uba'), Js('Uku'), Js('Ura'), Js('Ut'), Js('Zaki')]))
var.put('nm2', Js([Js('ba'), Js('bam'), Js('bani'), Js('bu'), Js('ha'), Js('hara'), Js('hu'), Js('ka'), Js('ku'), Js('lazu'), Js('lua'), Js('mea'), Js('nar'), Js('nara'), Js('naram'), Js('naru'), Js('nashtu'), Js('ni'), Js('niri'), Js('nu'), Js('nua'), Js('pana'), Js('ram'), Js('ranu'), Js('rashi'), Js('raya'), Js('ri'), Js('rin'), Js('runu'), Js('shara'), Js('shari'), Js('shi'), Js('shti'), Js('shtu'), Js('shu'), Js('sunu'), Js('ta'), Js('tana'), Js('tani'), Js('tari'), Js('ti'), Js('tira'), Js('tiru'), Js('tua'), Js('tum'), Js('wia'), Js('ya'), Js('yara'), Js('yua'), Js('zu')]))
pass
pass
pass


# Add lib to the module scope
dndShardmindNames = var.to_python()

File: js/dndShardmindNames.js
var nm1 = ["Adu","Ama","Ani","Ar","Arsha","Ashi","Ashtu","Bala","Bara","Basha","Beles","Delu","Di","Dura","Duru","Enu","Eri","Eshu","Hua","Hun","Il","Ilu","Ira","Ish","Ku","Kua","Kuba","Lu","Mani","Mara","Mashi","Na","Nara","Nashi","Nu","Rua","Run","Sana","Sari","Selu","Shir","Suma","Tab","Tin","Tiru","Uba","Uku","Ura","Ut","Zaki"];
var nm2 = ["ba","bam","bani","bu","ha","hara","hu","ka","ku","lazu","lua","mea","nar","nara","naram","naru","nashtu","ni","niri","nu","nua","pana","ram","ranu","rashi","raya","ri","rin","runu","shara","shari","shi","shti","shtu","shu","sunu","ta","tana","tani","tari","ti","tira","tiru","tua","tum","wia","ya","yara","yua","zu"];

function nameGen(){
	var br = "";
	var element = document.createElement("div");
	element.setAttribute("id", "result");
	
	for(i = 0; i < 10; i++){
		nameMas();
		while(nMs === ""){
			nameMas();
		}
		br = document.createElement('br');	
		element.appendChild(document.createTextNode(nMs));
		element.appendChild(br);
	}
	if(document.getElementById("result")){
		document.getElementById("placeholder").removeChild(document.getElementById("result"));
	}		
	document.getElementById("placeholder").appendChild(element);
}

function nameMas(){
	rnd = Math.floor(Math.random() * nm1.length);
	rnd2 = Math.floor(Math.random() * nm2.length);
	nMs = nm1[rnd] + nm2[rnd2];
	testSwear(nMs);
}